File: frontend/lib/models/user_preferences.dart
class UserPreferences {
  final String userId;
  final String? preferredName;
  final String communicationStyle;
  final String responseLength;
  final String interests;
  final String goals;
  final bool memoryEnabled;

  UserPreferences({
    required this.userId,
    this.preferredName,
    this.communicationStyle = 'casual',
    this.responseLength = 'moderate',
    this.interests = '[]',
    this.goals = '[]',
    this.memoryEnabled = true,
  });

  factory UserPreferences.fromJson(Map<String, dynamic> json) {
    return UserPreferences(
      userId: json['userId'] ?? '',
      preferredName: json['preferredName'],
      communicationStyle: json['communicationStyle'] ?? 'casual',
      responseLength: json['responseLength'] ?? 'moderate',
      interests: json['interests'] ?? '[]',
      goals: json['goals'] ?? '[]',
      memoryEnabled: json['memoryEnabled'] ?? true,
    );
  }
}

File: backend/src/main/java/com/innercircle/model/UserPreferences.java
package com.innercircle.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "user_preferences")
@Data
@NoArgsConstructor
public class UserPreferences {

    @Id
    @Column(name = "user_id")
    private UUID userId;

    private String preferredName;
    private String communicationStyle = "casual";
    private String responseLength = "moderate";

    @Column(columnDefinition = "TEXT DEFAULT '[]'")
    private String interests = "[]";

    @Column(columnDefinition = "TEXT DEFAULT '[]'")
    private String goals = "[]";

    private boolean memoryEnabled = true;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;
}
